src/filesystem.cc

```diff
@@ -893,7 +912,9 @@ const std::string AS_URL_PATTERN = "as://([^/]+)/([^/?]+)(?:/([^?]*))?(\\?.*)?";
 
 class ASFileSystem : public FileSystem {
  public:
-  ASFileSystem(const std::string& s3_path);
+  ASFileSystem(
+      const std::string& path, const std::string& account_str,
+      const std::string& account_key);
```

Comment:
Maybe encapsulate in a simple `AzureStorageOptions` struct?

Reply:
Updated.